src/test/java/org/opensearch/neuralsearch/common/BaseNeuralSearchIT.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.common;
+
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+import org.opensearch.client.*;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+
+public abstract class BaseNeuralSearchIT extends OpenSearchRestTestCase {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final Locale locale = Locale.getDefault();
+
+    public String uploadModel(String requestBody) throws Exception {
+        Response uploadResponse = makeRequest(
+            client(),
+            "POST",
+            "/_plugins/_ml/models/_upload",
+            null,
+            toHttpEntity(requestBody),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode uploadResJson = objectMapper.readTree(EntityUtils.toString(uploadResponse.getEntity()));
+        String taskId = uploadResJson.get("task_id").asText();
+        assertNotNull(taskId);
+
+        JsonNode taskQueryResult = getTaskQueryResponse(taskId);
+        boolean isComplete = checkComplete(taskQueryResult);
+        while (!isComplete) {
+            taskQueryResult = getTaskQueryResponse(taskId);
+            isComplete = checkComplete(taskQueryResult);
+        }
+        String modelId = taskQueryResult.get("model_id").asText();
+        assertNotNull(modelId);
+        return modelId;
+    }
+
+    public void loadModel(String modelId) throws IOException {
+        Response uploadResponse = makeRequest(
+            client(),
+            "POST",
+            String.format(locale, "/_plugins/_ml/models/%s/_load", modelId),
+            null,
+            toHttpEntity(""),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode uploadResJson = objectMapper.readTree(EntityUtils.toString(uploadResponse.getEntity()));
+        String taskId = uploadResJson.get("task_id").asText();
+        assertNotNull(taskId);
+
+        JsonNode taskQueryResult = getTaskQueryResponse(taskId);
+        boolean isComplete = checkComplete(taskQueryResult);
+        while (!isComplete) {
```

Comment:
Will this hang forever if load fails?

Reply:
No because failure loading task will return a result with error not empty. That said, added sleep to avoid CPU consuming and max time to limit the method running.